### PR TITLE
Fix completing-read prompter

### PR DESCRIPTION
**Pull request by minad**
There is a bug when switching to the completing-read prompter due to #287, fixed here.

I've also observed that when you press C-. C-h on a target at point the indicator will be shown in the long form before the minibuffer prompt "Act on ...". Did I introduce this or is this a regression due to #284?

### CODE REVIEW

File: embark.el [hunk 1]
```diff
@@ -751,23 +752,21 @@ UPDATE is the indicator update function."
                 (embark--read-key-sequence update)))
          (cmd (let ((overriding-terminal-local-map keymap))
                 (key-binding key))))
-    (setq cmd
```
Comment (oantolin) [inline embark.el:754]: This was silly, wasn't it? I think there's probably a historical reason it evolved to this, but I don't remember.

Comment (minad) [reply embark.el:754]: Embark evolved a lot

**Comment by oantolin** [comment]
> I've also observed that when you press C-. C-h on a target at point the indicator will be shown in the long form before the minibuffer prompt "Act on ...".

I'm not sure I follow. Are you talking about how the minimal indicator will always use the long form "Act on type(others) 'target'" if you are in the completing-read prompter? I did that on purpose, because the completing-read prompter clobbers the message that was in the echo area and if I didn't use the long form you wouldn't see what the target was or if there are other targets.

**Comment by oantolin** [comment]
Also, I seem to recall that was the behavior previously too. I seem to recall it was requested by someone who used the completing-read prompter by default.

**Comment by minad** [comment]
Hmm, this makes the prompt very long. Shall we go back to the short form now that we have target highlighting? We could make the short form slightly longer, writing "Act defun(...)" and show the ellipsis only if there are more targets. Maybe we can also make this configurable? I haven't checked how the configuration works currently for the minimal indicator.

**Comment by oantolin** [comment]
How about just adding a new line between the indicator string and the minibuffer prompt:
image

**Comment by oantolin** [comment]
I am a little reluctant to remove the target just because, as shown in that screenshot, there is no reason for it to coincide with the highlighted portion and its informative to know what the target really is.

**Comment by minad** [comment]
Somehow target highlighting is broken now with the verbose-indicator. But not with the minimal indicator. Do you observe this too?

**Comment by minad** [comment]
Weird, there is some buffer switching going on.

**Comment by oantolin** [comment]
I hadn't noticed. With the verbose indicator, initial highlighting is fine, but when you cycle it stop higlighting. Maybe a simple `save-window-excursion` is enough to fix this. I'll check.

**Comment by minad** [comment]
See my fix here #289

**Comment by oantolin** [comment]
I think that's a sensible fix for the issue you noticed about the length of the prompt: if the indicator is longer than 40% of the frame width it goes on its own line. How does that sound?

**Pull request merged**